cli/internal/fs/package_deps_hash.go

```diff
@@ -31,354 +21,352 @@ type PackageDepsOptions struct {
-		relativePath, err := filepath.Rel(relativeTo, filename)
+func GetHashableDeps(rootPath AbsolutePath, files []turbopath.AbsoluteSystemPath) (map[turbopath.RelativeUnixPath]string, error) {
+	output := make([]turbopath.RelativeUnixPath, len(files))
+	convertedRootPath := turbopath.AbsoluteSystemPath(rootPath.ToString())
+
+	for index, file := range files {
+		relativeSystemPath, err := file.Rel(convertedRootPath)
 		if err != nil {
-			return nil, errors.Wrapf(err, "failed to get relative path from %v to %v", relativeTo, relativePath)
+			return nil, err
 		}
-		key := filepath.ToSlash(relativePath)
-		result[key] = hash
+		output[index] = relativeSystemPath.ToRelativeUnixPath()
 	}
-	return result, nil
+	return gitHashObject(convertedRootPath, output)
 }
 
-// threadsafeBufferWriter is a wrapper around a byte buffer and a lock, allowing
-// multiple goroutines to write to the same buffer. No attempt is made to
-// lock around reading, which should only be done once no more writing will occur.
-type threadsafeBufferWriter struct {
-	mu     sync.Mutex
-	buffer bytes.Buffer
-}
+// gitHashObject returns a map of paths to their SHA hashes calculated by passing the paths `git hash-object`.
+// It will always accept a system path. On Windows it *also* accepts Unix paths.
+//
+// Note: relative paths passed to `git hash-object` are processed as relative to the *repository* root.
+// For that reason we convert all paths to absolute paths relative to the rootPath prior to passing them
+// to `git hash-object`.
+func gitHashObject(rootPath turbopath.AbsoluteSystemPath, filesToHash []turbopath.RelativeUnixPath) (map[turbopath.RelativeUnixPath]string, error) {
+	fileCount := len(filesToHash)
+	output := make(map[turbopath.RelativeUnixPath]string, fileCount)
+
+	if fileCount > 0 {
+		cmd := exec.Command(
+			"git",           // Using `git` from $PATH,
+			"hash-object",   // hash a file,
+			"--stdin-paths", // using a list of newline-separated paths from stdin.
+		)
+		cmd.Dir = rootPath.ToString() // Start at this directory.
+
+		// The functionality for gitHashObject is different enough that it isn't reasonable to
+		// generalize the behavior for `runGitCmd`. In fact, it doesn't even use the `gitoutput`
+		// encoding library, instead relying on its own separate `bufio.Scanner`.
+
+		// We're going to send the list of files in via `stdin`, so we grab that pipe.
+		// This prevents a huge number of encoding issues and shell compatibility issues
+		// before they even start.
+		stdinPipe, stdinPipeError := cmd.StdinPipe()
+		if stdinPipeError != nil {
+			return nil, stdinPipeError
+		}
 
-func (tsbw *threadsafeBufferWriter) Write(p []byte) (int, error) {
-	tsbw.mu.Lock()
-	defer tsbw.mu.Unlock()
-	return tsbw.buffer.Write(p)
-}
+		// Kick the processing off in a goroutine so while that is doing its thing we can go ahead
+		// and wire up the consumer of `stdout`.
+		go func() {
+			defer func() {
+				stdinPipeCloseError := stdinPipe.Close()
```

Comment:
This ceremony is because of this: https://github.com/kisielk/errcheck/issues/101

The maintainer of errcheck is focusing on correctness, not pragmatism.

Reply:
I've added an alternative optional commit 4da42d81813066f67ea9487a4aa56ce9481290da which copypasta's a standard `golangci-lint` rule to ignore this.

Can either do the `defer func()` ceremony or configure the linter this way. Your call.